content/ADR-213-quests-ui.md

```diff
@@ -0,0 +1,55 @@
+
+## Solution Space Exploration
+
+There are three potential solutions for designing the Quests System UI:
+
+Using pure SDK components: In this approach, we would create the UI using only the components provided by the SDK. While this approach provides consistency across implementations, there may be some gaps that we would need to fill by implementing additional components.
+
+Using SDK to communicate the UI state to the Explorer: In this approach, we would use the SDK to store the UI state and communicate it to the Explorer. This would reduce the amount of communication required between the Explorer and the backend to retrieve information needed to render the UI. However, we would need to ensure that the SDK's capabilities are sufficient for our use case.
+
+Using the Explorer for everything: In this approach, each Explorer implementation would be responsible for implementing the entire UI. While this approach provides the most flexibility, it also means that each implementation would need to re-do everything, which could result in inconsistent UIs across different implementations.
+
+Each of these approaches has its benefits and drawbacks, and we need to carefully consider them before making a decision.
+
+## Specification
+
+The technical specification for this approach will involve defining the specific data storage mechanisms that will be used in the SDK, as well as the frontend rendering logic. It will also require integration between the two systems to ensure that they are communicating effectively. The UI design will need to be developed in parallel with the technical implementation.
+
+Based on the mockups we have, the Quests System UI should include the following components:
+
+Quests HUD: A list of active quests with current tasks. This component should provide a quick overview of the user's progress in each quest.
+
+Quests Log: A window containing all the details for each quest, including name, description, steps, and rewards. Users should be able to view their progress and access information about each quest, and also be able to abandon a quest by clicking a button in this UI.
+
+New Quests Offer: A pop-up with a proposal to start a new quest, with buttons to accept or decline. This component should make it easy for users to start a new quest and ensure they don't miss any available quests.
+
+Step Completed Popup: Whenever a quest step is completed, a feedback or animation should be shown to indicate progress and keep the user engaged.
+
+## Decision
+
+After exploring the available options, we have decided to use the Explorer for everything and the first option (develop new SDK components) will be considered as a nice-to-have in the future. This decision was made based on the fact that using the Explorer for everything provides more flexibility and control over the UI.
```

Comment:
```suggestion
After exploring the available options, we have decided to use the Explorer for everything. It would be nice to have the first option (develop new SDK components), so it will be considered in the future. The decision is based on the fact that using the Explorer for everything provides more flexibility and control over the UI.
```